go-controller/pkg/ovn/gatewayrouter/policybasedroutes.go

```diff
@@ -0,0 +1,209 @@
+	}
+
+	return nil
+}
+
+// This function syncs logical router policies given various criteria
+// This function compares the following ovn-nbctl output:
+
+// either
+
+// 		72db5e49-0949-4d00-93e3-fe94442dd861,ip4.src == 10.244.0.2 && ip4.dst == 172.18.0.2 /* ovn-worker2 */,169.254.0.1
+// 		6465e223-053c-4c74-a5f0-5f058c9f7a3e,ip4.src == 10.244.2.2 && ip4.dst == 172.18.0.3 /* ovn-worker */,169.254.0.1
+// 		7debdcc6-ad5e-4825-9978-74bfbf8b7c27,ip4.src == 10.244.1.2 && ip4.dst == 172.18.0.4 /* ovn-control-plane */,169.254.0.1
+
+// or
+
+// 		c20ac671-704a-428a-a32b-44da2eec8456,"inport == ""rtos-ovn-worker2"" && ip4.dst == 172.18.0.2 /* ovn-worker2 */",10.244.0.2
+// 		be7c8b53-f8ac-4051-b8f1-bfdb007d0956,"inport == ""rtos-ovn-worker"" && ip4.dst == 172.18.0.3 /* ovn-worker */",10.244.2.2
+// 		fa8cf55d-a96c-4a53-9bf2-1c1fb1bc7a42,"inport == ""rtos-ovn-control-plane"" && ip4.dst == 172.18.0.4 /* ovn-control-plane */",10.244.1.2
+
+// or
+
+// 		822ab242-cce5-47b2-9c6f-f025f47e766a,ip4.src == 10.244.2.2  && ip4.dst != 10.244.0.0/16 /* inter-ovn-worker */,169.254.0.1
+// 		a1b876f6-5ed4-4f88-b09c-7b4beed3b75f,ip4.src == 10.244.1.2  && ip4.dst != 10.244.0.0/16 /* inter-ovn-control-plane */,169.254.0.1
+// 		0f5af297-74c8-4551-b10e-afe3b74bb000,ip4.src == 10.244.0.2  && ip4.dst != 10.244.0.0/16 /* inter-ovn-worker2 */,169.254.0.1
+
+// The function checks to see if the mgmtPort IP has changed, or if match criteria has changed
+// and removes stale policies for a node for the NodeSubnetPolicy in SGW.
+// TODO: Fix the MGMTPortPolicy's and InterNodePolicy's ip4.src fields if the mgmtPort IP has changed in LGW.
+// It also adds new policies for a node at a specific priority.
+// This is ugly (since the node is encoded as a comment in the match),
+// but a necessary evil as any change to this would break upgrades and
+// possible downgrades. We could make sure any upgrade encodes the node in
+// the external_id, but since ovn-kubernetes isn't versioned, we won't ever
+// know which version someone is running of this and when the switch to version
+// N+2 is fully made.
+func (pbr *PolicyBasedRoutesManager) sync(nodeName string, matches sets.Set[string], priority, nexthop string) error {
+	// create a map to track matches found
+	matchTracker := sets.New(sets.List(matches)...)
+
+	if priority == ovntypes.NodeSubnetPolicyPriority {
+		policies, err := pbr.findPolicyBasedRoutes(priority)
+		if err != nil {
+			return fmt.Errorf("unable to list policies, err: %v", err)
+		}
+
+		// sync and remove unknown policies for this node/priority
+		// also flag if desired policies are already found
+		for _, policy := range policies {
+			if strings.Contains(policy.Match, fmt.Sprintf("%s\"", nodeName)) {
+				// if the policy is for this node and has the wrong mgmtPortIP as nexthop, remove it
+				// FIXME we currently assume that foundNexthops is a single ip, this may
+				// change in the future.
+
```

Comment:
maybe we just `continue` if theres no next hops actually - trying not to add another len() to ln 116

Reply:
done, theres validation for next hops